packages/data/src/test/privateAPIs.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Internal dependencies
+ */
+import { createRegistry } from '../registry';
+import createReduxStore from '../redux-store';
+import { unlock } from '../experiments';
+
+/**
+ * WordPress dependencies
+ */
+
+beforeEach( () => {
+	jest.useFakeTimers( 'legacy' );
+} );
+
+afterEach( () => {
+	jest.runOnlyPendingTimers();
+	jest.useRealTimers();
+} );
+
+describe( 'Private data APIs', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createRegistry();
+	} );
+
+	function getPublicPrice( state ) {
+		return state.price;
+	}
+	function getSecretDiscount( state ) {
+		return state.secretDiscount;
+	}
+	function setSecretDiscount( price ) {
+		return { type: 'SET_PRIVATE_PRICE', price };
+	}
+
+	function setPublicPrice( price ) {
+		return { type: 'SET_PUBLIC_PRICE', price };
+	}
+	function createStore() {
+		const groceryStore = createReduxStore( 'grocer', {
+			selectors: {
+				getPublicPrice,
+				getState: ( state ) => state,
+			},
+			actions: { setPublicPrice },
+			reducer: ( state, action ) => {
+				if ( action?.type === 'SET_PRIVATE_PRICE' ) {
+					return {
+						...state,
+						secretDiscount: action?.price,
+					};
+				}
+				return {
+					price: 1000,
+					secretDiscount: 800,
+					...( state || {} ),
+				};
+			},
+		} );
+		registry.register( groceryStore );
+		return groceryStore;
+	}
+
+	describe( 'private selectors', () => {
+		it( 'should expose public selectors by default', () => {
+			const groceryStore = createStore();
+			unlock( groceryStore ).registerPrivateSelectors( groceryStore, {
+				getSecretDiscount,
+			} );
+
+			const publicSelectors = registry.select( groceryStore );
+			expect( publicSelectors.getPublicPrice ).toEqual(
+				expect.any( Function )
+			);
+		} );
+
+		it( 'should not expose private selectors by default', () => {
+			const groceryStore = createStore();
+			unlock( groceryStore ).registerPrivateSelectors( {
+				getSecretDiscount,
+			} );
+
+			const publicSelectors = registry.select( groceryStore );
+			expect( publicSelectors.getSecretDiscount ).toEqual( undefined );
+		} );
+
+		it( 'should make private selectors available via unlock()', () => {
+			const groceryStore = createStore();
+			unlock( groceryStore ).registerPrivateSelectors( {
+				getSecretDiscount,
+			} );
+
+			const privateSelectors = unlock( registry.select( groceryStore ) );
+			expect( privateSelectors.getSecretDiscount ).toEqual(
```

Comment:
Might be good to assert here that the public selector still exists on the object returned by `unlock()`.

Reply:
They didn't! So good call – I pushed a remedy in 954b353efe.